src/DocumentFormat.OpenXml/Framework/Schema/ParticlePath.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DocumentFormat.OpenXml.Validation.Schema;
+using System;
+
+namespace DocumentFormat.OpenXml.Framework.Schema
+{
+    internal class ParticlePath : IComparable<ParticlePath>, IEquatable<ParticlePath>
+    {
+        private readonly ParticlePathItem[] _values;
+
+        public static ParticlePath Create(params ParticlePathItem[] values)
+            => new ParticlePath(values);
+
+        private ParticlePath(ParticlePathItem[] values)
+        {
+            _values = values;
+        }
+
+        public ParticleType Type => _values[_values.Length - 1].Type;
+
+        public int CompareTo(ParticlePath other)
+            => CompareTo(other, true);
+
+#if DEBUG
+        public string Path => string.Join(", ", (object[])_values);
+#endif
+
+        private int CompareTo(ParticlePath other, bool isCompare)
+        {
+            if (other is null)
+            {
+                return -1;
+            }
+
+            var en = new Enumerator(this);
+            var on = new Enumerator(other);
+
+            while (en.MoveNext() && on.MoveNext())
+            {
+                var (currentParticleType, currentIndex) = en.Current;
+                var (otherParticleType, otherIndex) = on.Current;
+
+                if (currentParticleType == ParticleType.Element && otherParticleType == ParticleType.Element)
+                {
+                    return currentIndex.CompareTo(otherIndex);
+                }
+                else if (currentParticleType == ParticleType.Sequence && otherParticleType == ParticleType.Sequence)
+                {
+                    var compared = currentIndex.CompareTo(otherIndex);
+
+                    if (compared != 0)
+                    {
+                        return compared;
+                    }
+                }
+                else if (currentParticleType == ParticleType.Choice && otherParticleType == ParticleType.Choice)
+                {
+                    if (currentIndex != otherIndex)
```

Comment:
I think I see why indexes being different here returns 0 (because choices are exclusive and replace each other), but it may be worth adding a comment to make it clear since currentIndex != otherIndex => 0 is weird as far as comparers go.

Reply:
Yes, it is a little weird. I added a comment